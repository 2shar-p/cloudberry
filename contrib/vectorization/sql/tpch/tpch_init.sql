DROP TABLE IF EXISTS nation;
DROP TABLE IF EXISTS region;
DROP TABLE IF EXISTS part;
DROP TABLE IF EXISTS supplier;
DROP TABLE IF EXISTS partsupp;
DROP TABLE IF EXISTS customer;
DROP TABLE IF EXISTS orders;
DROP TABLE IF EXISTS lineitem;

set default_table_access_method=ao_column;


CREATE TABLE nation (
    N_NATIONKEY  INTEGER NOT NULL,
    N_NAME       TEXT NOT NULL,
    N_REGIONKEY  INTEGER NOT NULL,
    N_COMMENT    VARCHAR(152));

\copy nation FROM 'data/tpch/nation.csv'  DELIMITER ',' CSV;


CREATE TABLE region (
    R_REGIONKEY  INTEGER NOT NULL,
    R_NAME       TEXT NOT NULL,
    R_COMMENT    VARCHAR(152));

\copy region FROM 'data/tpch/region.csv'  DELIMITER ',' CSV;


CREATE TABLE part (
    P_PARTKEY     INTEGER NOT NULL,
    P_NAME        VARCHAR(55) NOT NULL,
    P_MFGR        TEXT NOT NULL,
    P_BRAND       TEXT NOT NULL,
    P_TYPE        VARCHAR(25) NOT NULL,
    P_SIZE        INTEGER NOT NULL,
    P_CONTAINER   TEXT NOT NULL,
    P_RETAILPRICE DECIMAL(15,2) NOT NULL,
    P_COMMENT     VARCHAR(23) NOT NULL);

\copy part FROM 'data/tpch/part.csv'  DELIMITER ',' CSV;


CREATE TABLE supplier (
    S_SUPPKEY     INTEGER NOT NULL,
    S_NAME        TEXT NOT NULL,
    S_ADDRESS     VARCHAR(40) NOT NULL,
    S_NATIONKEY   INTEGER NOT NULL,
    S_PHONE       TEXT NOT NULL,
    S_ACCTBAL     DECIMAL(15,2) NOT NULL,
    S_COMMENT     VARCHAR(101) NOT NULL);

\copy supplier FROM 'data/tpch/supplier.csv'  DELIMITER ',' CSV;


CREATE TABLE partsupp (
    PS_PARTKEY     INTEGER NOT NULL,
    PS_SUPPKEY     INTEGER NOT NULL,
    PS_AVAILQTY    INTEGER NOT NULL,
    PS_SUPPLYCOST  DECIMAL(15,2)  NOT NULL,
    PS_COMMENT     VARCHAR(199) NOT NULL );

\copy partsupp FROM 'data/tpch/partsupp.csv'  DELIMITER ',' CSV;


CREATE TABLE customer (
    C_CUSTKEY     INTEGER NOT NULL,
    C_NAME        VARCHAR(25) NOT NULL,
    C_ADDRESS     VARCHAR(40) NOT NULL,
    C_NATIONKEY   INTEGER NOT NULL,
    C_PHONE       TEXT NOT NULL,
    C_ACCTBAL     DECIMAL(15,2)   NOT NULL,
    C_MKTSEGMENT  TEXT NOT NULL,
    C_COMMENT     VARCHAR(117) NOT NULL);

\copy customer FROM 'data/tpch/customer.csv'  DELIMITER ',' CSV;


CREATE TABLE orders (
    O_ORDERKEY       INT8 NOT NULL,
    O_CUSTKEY        INTEGER NOT NULL,
    O_ORDERSTATUS    TEXT NOT NULL,
    O_TOTALPRICE     DECIMAL(15,2) NOT NULL,
    O_ORDERDATE      DATE NOT NULL,
    O_ORDERPRIORITY  TEXT NOT NULL,
    O_CLERK          TEXT NOT NULL,
    O_SHIPPRIORITY   INTEGER NOT NULL,
    O_COMMENT        VARCHAR(79) NOT NULL);

\copy orders FROM 'data/tpch/orders.csv'  DELIMITER ',' CSV;

CREATE TABLE lineitem (
    L_ORDERKEY    INT8 NOT NULL,
    L_PARTKEY     INTEGER NOT NULL,
    L_SUPPKEY     INTEGER NOT NULL,
    L_LINENUMBER  INTEGER NOT NULL,
    L_QUANTITY    DECIMAL(15,2) NOT NULL,
    L_EXTENDEDPRICE  DECIMAL(15,2) NOT NULL,
    L_DISCOUNT    DECIMAL(15,2) NOT NULL,
    L_TAX         DECIMAL(15,2) NOT NULL,
    L_RETURNFLAG  TEXT NOT NULL,
    L_LINESTATUS  TEXT NOT NULL,
    L_SHIPDATE    DATE NOT NULL,
    L_COMMITDATE  DATE NOT NULL,
    L_RECEIPTDATE DATE NOT NULL,
    L_SHIPINSTRUCT TEXT NOT NULL,
    L_SHIPMODE     TEXT NOT NULL,
    L_COMMENT      VARCHAR(44) NOT NULL);


\copy lineitem FROM 'data/tpch/lineitem.csv'  DELIMITER ',' CSV;
