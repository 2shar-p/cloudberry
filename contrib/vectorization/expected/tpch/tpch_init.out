DROP TABLE IF EXISTS nation;
NOTICE:  table "nation" does not exist, skipping
DROP TABLE IF EXISTS region;
NOTICE:  table "region" does not exist, skipping
DROP TABLE IF EXISTS part;
NOTICE:  table "part" does not exist, skipping
DROP TABLE IF EXISTS supplier;
NOTICE:  table "supplier" does not exist, skipping
DROP TABLE IF EXISTS partsupp;
NOTICE:  table "partsupp" does not exist, skipping
DROP TABLE IF EXISTS customer;
NOTICE:  table "customer" does not exist, skipping
DROP TABLE IF EXISTS orders;
NOTICE:  table "orders" does not exist, skipping
DROP TABLE IF EXISTS lineitem;
NOTICE:  table "lineitem" does not exist, skipping
set default_table_access_method=ao_column;
CREATE TABLE nation (
    N_NATIONKEY  INTEGER NOT NULL,
    N_NAME       TEXT NOT NULL,
    N_REGIONKEY  INTEGER NOT NULL,
    N_COMMENT    VARCHAR(152));
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'n_nationkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy nation FROM 'data/tpch/nation.csv'  DELIMITER ',' CSV;
CREATE TABLE region (
    R_REGIONKEY  INTEGER NOT NULL,
    R_NAME       TEXT NOT NULL,
    R_COMMENT    VARCHAR(152));
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'r_regionkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy region FROM 'data/tpch/region.csv'  DELIMITER ',' CSV;
CREATE TABLE part (
    P_PARTKEY     INTEGER NOT NULL,
    P_NAME        VARCHAR(55) NOT NULL,
    P_MFGR        TEXT NOT NULL,
    P_BRAND       TEXT NOT NULL,
    P_TYPE        VARCHAR(25) NOT NULL,
    P_SIZE        INTEGER NOT NULL,
    P_CONTAINER   TEXT NOT NULL,
    P_RETAILPRICE FLOAT8 NOT NULL,
    P_COMMENT     VARCHAR(23) NOT NULL);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'p_partkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy part FROM 'data/tpch/part.csv'  DELIMITER ',' CSV;
CREATE TABLE supplier (
    S_SUPPKEY     INTEGER NOT NULL,
    S_NAME        TEXT NOT NULL,
    S_ADDRESS     VARCHAR(40) NOT NULL,
    S_NATIONKEY   INTEGER NOT NULL,
    S_PHONE       TEXT NOT NULL,
    S_ACCTBAL     FLOAT8 NOT NULL,
    S_COMMENT     VARCHAR(101) NOT NULL);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 's_suppkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy supplier FROM 'data/tpch/supplier.csv'  DELIMITER ',' CSV;
CREATE TABLE partsupp (
    PS_PARTKEY     INTEGER NOT NULL,
    PS_SUPPKEY     INTEGER NOT NULL,
    PS_AVAILQTY    INTEGER NOT NULL,
    PS_SUPPLYCOST  FLOAT8  NOT NULL,
    PS_COMMENT     VARCHAR(199) NOT NULL );
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'ps_partkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy partsupp FROM 'data/tpch/partsupp.csv'  DELIMITER ',' CSV;
CREATE TABLE customer (
    C_CUSTKEY     INTEGER NOT NULL,
    C_NAME        VARCHAR(25) NOT NULL,
    C_ADDRESS     VARCHAR(40) NOT NULL,
    C_NATIONKEY   INTEGER NOT NULL,
    C_PHONE       TEXT NOT NULL,
    C_ACCTBAL     FLOAT8   NOT NULL,
    C_MKTSEGMENT  TEXT NOT NULL,
    C_COMMENT     VARCHAR(117) NOT NULL);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'c_custkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy customer FROM 'data/tpch/customer.csv'  DELIMITER ',' CSV;
CREATE TABLE orders (
    O_ORDERKEY       INT8 NOT NULL,
    O_CUSTKEY        INTEGER NOT NULL,
    O_ORDERSTATUS    TEXT NOT NULL,
    O_TOTALPRICE     FLOAT8 NOT NULL,
    O_ORDERDATE      DATE NOT NULL,
    O_ORDERPRIORITY  TEXT NOT NULL,
    O_CLERK          TEXT NOT NULL,
    O_SHIPPRIORITY   INTEGER NOT NULL,
    O_COMMENT        VARCHAR(79) NOT NULL);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'o_orderkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy orders FROM 'data/tpch/orders.csv'  DELIMITER ',' CSV;
CREATE TABLE lineitem (
    L_ORDERKEY    INT8 NOT NULL,
    L_PARTKEY     INTEGER NOT NULL,
    L_SUPPKEY     INTEGER NOT NULL,
    L_LINENUMBER  INTEGER NOT NULL,
    L_QUANTITY    FLOAT8 NOT NULL,
    L_EXTENDEDPRICE  FLOAT8 NOT NULL,
    L_DISCOUNT    FLOAT8 NOT NULL,
    L_TAX         FLOAT8 NOT NULL,
    L_RETURNFLAG  TEXT NOT NULL,
    L_LINESTATUS  TEXT NOT NULL,
    L_SHIPDATE    DATE NOT NULL,
    L_COMMITDATE  DATE NOT NULL,
    L_RECEIPTDATE DATE NOT NULL,
    L_SHIPINSTRUCT TEXT NOT NULL,
    L_SHIPMODE     TEXT NOT NULL,
    L_COMMENT      VARCHAR(44) NOT NULL);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'l_orderkey' as the Cloudberry Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
\copy lineitem FROM 'data/tpch/lineitem.csv'  DELIMITER ',' CSV;
